fabric-events-lifecycle-v0/src/main/java/net/fabricmc/fabric/mixin/eventslifecycle/MixinEntityType.java

```diff
@@ -0,0 +1,42 @@
+package net.fabricmc.fabric.mixin.eventslifecycle;
+
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.fabricmc.fabric.api.event.entity.EntityTickCallback;
+import net.fabricmc.fabric.impl.event.EntityTypeCaller;
+import net.minecraft.entity.Entity;
+import net.minecraft.entity.EntityType;
+import net.minecraft.util.profiler.Profiler;
+import org.spongepowered.asm.mixin.Mixin;
+
+@Mixin(EntityType.class)
+public class MixinEntityType<T extends Entity> implements EntityTypeCaller {
+	private EntityTickCallback event;
+
+	@Override
+	public EntityTickCallback getEntityEvent() {
+		if (event == null) {
+			event = (EntityTickCallback<T>)EventFactory.createArrayBacked(EntityTickCallback.class,
```

Comment:
Somehow a fairly bad crash slipped into this PR. Makes me think those test mods should really get fixed...